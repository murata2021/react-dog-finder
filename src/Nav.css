.Nav {
    padding: 2rem;
    background-color: blue;
}

.Nav a{
    margin: 1rem;
    color: white;
    text-decoration-line: none;
    justify-content: center;

}