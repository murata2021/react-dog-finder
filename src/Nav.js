import React from "react";
import { NavLink } from "react-router-dom";
import './Nav.css'


const Nav=({dogs})=>{

    console.log(dogs)

    return (
        <nav className="Nav">
            {dogs.map(dog=><NavLink to={`/dogs/${dog.name}`}>{dog.name}</NavLink>)}
        </nav>
    )

}

export default Nav;