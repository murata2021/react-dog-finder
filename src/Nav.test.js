import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Nav from './Nav';

const initialDogList={
    dogs: [
      {
        name: "Whiskey",
        age: 5,
        src: "",
        facts: [
          "Whiskey loves eating popcorn.",
          "Whiskey is a terrible guard dog.",
          "Whiskey wants to cuddle with you!"
        ]
      },
      {
        name: "Duke",
        age: 3,
        src: "",
        facts: [
          "Duke believes that ball is life.",
          "Duke likes snow.",
          "Duke enjoys pawing other dogs."
        ]
      },
      {
        name: "Perry",
        age: 4,
        src: "",
        facts: [
          "Perry loves all humans.",
          "Perry demolishes all snacks.",
          "Perry hates the rain."
        ]
      },
      {
        name: "Tubby",
        age: 4,
        src: "",
        facts: [
          "Tubby is really stupid.",
          "Tubby does not like walks.",
          "Angelina used to hate Tubby, but claims not to anymore."
        ]
      }
    ]
  }

it('renders without crashing',()=>{

    render(<MemoryRouter><Nav dogs={initialDogList.dogs}/></MemoryRouter>)
})

it("matches snapshot", function() {
    const {asFragment} = render(<MemoryRouter><Nav dogs={initialDogList.dogs}/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });

it('renders with correct elements',()=>{

    const {debug,getByText,queryByText}=render(<MemoryRouter><Nav dogs={initialDogList.dogs}/></MemoryRouter>)
    expect(getByText('Whiskey')).toBeInTheDocument()
    expect(getByText('Duke')).toBeInTheDocument()
    expect(getByText('Perry')).toBeInTheDocument()
    expect(getByText('Tubby')).toBeInTheDocument()

    expect(queryByText('Snoopy')).not.toBeInTheDocument()

})