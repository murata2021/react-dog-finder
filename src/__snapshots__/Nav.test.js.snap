// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Nav"
  >
    <a
      href="/dogs/Whiskey"
    >
      Whiskey
    </a>
    <a
      href="/dogs/Duke"
    >
      Duke
    </a>
    <a
      href="/dogs/Perry"
    >
      Perry
    </a>
    <a
      href="/dogs/Tubby"
    >
      Tubby
    </a>
  </nav>
</DocumentFragment>
`;
